File: src/main/java/com/skocur/imagecipher/plugin/lib/IcEncrypter.java
package com.skocur.imagecipher.plugin.lib;

import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;

public abstract class IcEncrypter implements AutoCloseable {
    protected String fileName;
    protected BufferedImage image;

    public abstract void encrypt(String var1);

    public String getFileName() {
        return this.fileName;
    }

    public void setSourceFile(String fileName) throws IOException {
        this.fileName = fileName;
        this.image = ImageIO.read(new File(fileName));
    }

    public BufferedImage getImage() {
        return this.image;
    }

    public void setImage(BufferedImage image) {
        this.image = image;
    }
}

File: app/src/main/kotlin/com/imagecipher/app/plugin/lib/IcEncrypter.kt
package com.imagecipher.app.plugin.lib

import java.awt.image.BufferedImage
import java.io.File
import java.io.IOException
import javax.imageio.ImageIO

abstract class IcEncrypter : AutoCloseable {
    private var fileName: String? = null
    private var image: BufferedImage? = null

    abstract fun encrypt(text: String)

    fun getFileName(): String? {
        return fileName
    }

    @Throws(IOException::class)
    fun setSourceFile(fileName: String) {
        this.fileName = fileName
        this.image = ImageIO.read(File(fileName))
    }

    fun getImage(): BufferedImage? {
        return image
    }

    fun setImage(image: BufferedImage) {
        this.image = image
    }
}
